device.go

```diff
@@ -170,11 +174,71 @@ func newBlockDevice(devInfo DeviceInfo) *BlockDevice {
 	}
 }
 
+func makeBlockDevIDForHypervisor(deviceID string) string {
+	devID := fmt.Sprintf("drive-%s", deviceID)
+	if len(devID) > maxDevIDSize {
+		devID = string(devID[:maxDevIDSize])
```

Comment:
`maxDevIDSize` is specific to qemu and that's the reason why I wanted this check to be implemented in qemu.go through a new interface function of the hypervisor (something like capabilities() or so). But I think we can do that in a follow up PR since I don't want to block this one forever.